src/CoreWCF.Primitives/src/CoreWCF/Telemetry/AssemblyVersionExtensions.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Reflection;
+
+namespace CoreWCF.Telemetry;
+
+internal static class AssemblyVersionExtensions
+{
+    public static string GetPackageVersion(this Assembly assembly)
+    {
+        // MinVer https://github.com/adamralph/minver?tab=readme-ov-file#version-numbers
+        // together with Microsoft.SourceLink.GitHub https://github.com/dotnet/sourcelink
+        // fills AssemblyInformationalVersionAttribute by
+        // {majorVersion}.{minorVersion}.{patchVersion}.{pre-release label}.{pre-release version}.{gitHeight}+{Git SHA of current commit}
+        // Ex: 1.5.0-alpha.1.40+807f703e1b4d9874a92bd86d9f2d4ebe5b5d52e4
+        // The following parts are optional: pre-release label, pre-release version, git height, Git SHA of current commit
+        // For package version, value of AssemblyInformationalVersionAttribute without commit hash is returned.
+
+        var informationalVersion = assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion;
```

Comment:
Whenever possible, I prefer to use GetCustomAttributesData to get the attribute. It doesn't require instantiating the attribute itself.

Reply:
Do you think it is worth changing it? Should it be faster, or should we benchmark this for the best solution?